notebooks/data_loader.ipynb

```
@@ -0,0 +1,147 @@
+      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[4]\u001b[39m\u001b[32m, line 9\u001b[39m\n\u001b[32m      5\u001b[39m \n\u001b[32m      6\u001b[39m \u001b[38;5;66;03m# Setup logging\u001b[39;00m\n\u001b[32m      7\u001b[39m logging.basicConfig(level=logging.INFO, format=\u001b[33m'%(levelname)s: %(message)s'\u001b[39m)\n\u001b[32m      8\u001b[39m logger = logging.getLogger(__name__)\n\u001b[32m----> \u001b[39m\u001b[32m9\u001b[39m BASE_DIR = os.path.dirname(os.path.abspath(__file__))\n\u001b[32m     10\u001b[39m DEFAULT_CACHE_PATH = os.path.join(BASE_DIR, \u001b[33m'data'\u001b[39m, \u001b[33m'beijing_aqi.csv'\u001b[39m)\n",
+      "\u001b[31mNameError\u001b[39m: name '__file__' is not defined"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import logging\n",
+    "import pandas as pd\n",
+    "from ucimlrepo import fetch_ucirepo\n",
+    "\n",
+    "# Setup logging\n",
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)